packages/server/src/sdk/app/views/index.ts

```diff
@@ -299,15 +292,24 @@ export async function enrichSchema(
 
   const viewSchema = view.schema || {}
   const anyViewOrder = Object.values(viewSchema).some(ui => ui.order != null)
-  for (const key of Object.keys(tableSchema).filter(
-    k => tableSchema[k].visible !== false
-  )) {
+
+  const visibleSchemaFields = Object.keys(viewSchema).filter(key => {
+    if (helpers.views.isCalculationField(viewSchema[key])) {
+      return viewSchema[key].visible !== false
+    }
+    return key in tableSchema && tableSchema[key].visible !== false
+  })
+  const visibleTableFields = Object.keys(tableSchema).filter(
+    key => tableSchema[key].visible !== false
+  )
+  const visibleFields = new Set([...visibleSchemaFields, ...visibleTableFields])
+  for (const key of visibleFields) {
     // if nothing specified in view, then it is not visible
     const ui = viewSchema[key] || { visible: false }
     schema[key] = {
-      ...tableSchema[key],
+      ...(tableSchema[key] || {}),
```

Comment:
Is this required? Spread should work even if it is undefined

Reply:
Huh, I didn't know this. I assumed it wouldn't work. Will put this back how it was!